doc/dev/model.md

```diff
@@ -0,0 +1,78 @@
+# Overview
+
+This document provides an introduction to the msrest model and dpg model architectures.
```

Comment:
If we're planning to put that into a ChangeLog for customers, we shouldn't use internal wording.
```suggestion
This document provides an explanation on the changes applied to model structure for our newest generation of SDK.
```